gems/sorbet-runtime/test/types/attached_class.rb

```diff
@@ -67,5 +67,20 @@ def self.make
 
       assert_equal(Base.make.class, Integer)
     end
+
+    it 'can print its own name' do
+      class Base
+        extend T::Sig
+
+        sig {returns(T.attached_class)}
+        def self.make
+          Base.new
+        end
+      end
+
+      meth = Base.method(:make)
+      name = T::Private::Methods.signature_for_method(meth).return_type.to_s
+      assert_equal("T.attached_class", name)
```

Comment:
ooc, why not

```ruby
assert_equal("T.attached_class", T.attached_class.name)
```

?

Reply:
This is how we use it trough `tapioca`, I wanted to make sure we would get a failing test if this value changed.